src/mito2/src/read/seq_scan.rs

```diff
@@ -525,292 +290,66 @@ impl RegionScanner for SeqScan {
-        // No need to merge parts.
-        return parts;
-    }
-
-    // Sort parts by number of memtables and ranges in reverse order.
-    parts.sort_unstable_by(|a, b| {
-        a.memtable_ranges
-            .len()
-            .cmp(&b.memtable_ranges.len())
-            .then_with(|| {
-                let a_ranges_len = a
-                    .file_ranges
-                    .iter()
-                    .map(|ranges| ranges.len())
-                    .sum::<usize>();
-                let b_ranges_len = b
-                    .file_ranges
-                    .iter()
-                    .map(|ranges| ranges.len())
-                    .sum::<usize>();
-                a_ranges_len.cmp(&b_ranges_len)
-            })
-            .reverse()
-    });
-
-    let parts_to_reduce = parts.len() - parallelism;
-    for _ in 0..parts_to_reduce {
-        // Safety: We ensure `parts.len() > parallelism`.
-        let part = parts.pop().unwrap();
-        parts.last_mut().unwrap().merge(part);
-    }
-
-    parts
-}
-
-/// Splits parts by parallelism.
-/// It splits a part if it only scans one file and doesn't scan any memtable.
-fn maybe_split_parts(mut parts: Vec<ScanPart>, parallelism: usize) -> Vec<ScanPart> {
-    assert!(parallelism > 0);
-    if parts.len() >= parallelism {
-        // No need to split parts.
-        return parts;
-    }
-
-    let has_part_to_split = parts.iter().any(|part| part.can_split_preserve_order());
-    if !has_part_to_split {
-        // No proper parts to scan.
-        return parts;
-    }
-
-    // Sorts parts by the number of ranges in the first file.
-    parts.sort_unstable_by(|a, b| {
-        let a_len = a.file_ranges.first().map(|file| file.len()).unwrap_or(0);
-        let b_len = b.file_ranges.first().map(|file| file.len()).unwrap_or(0);
-        a_len.cmp(&b_len).reverse()
-    });
-    let num_parts_to_split = parallelism - parts.len();
-    let mut output_parts = Vec::with_capacity(parallelism);
-    // Split parts up to num_parts_to_split.
-    for part in parts.iter_mut() {
-        if !part.can_split_preserve_order() {
-            continue;
-        }
-        // Safety: `can_split_preserve_order()` ensures file_ranges.len() == 1.
-        // Splits part into `num_parts_to_split + 1` new parts if possible.
-        let target_part_num = num_parts_to_split + 1;
-        let ranges_per_part = (part.file_ranges[0].len() + target_part_num - 1) / target_part_num;
-        // `can_split_preserve_order()` ensures part.file_ranges[0].len() > 1.
-        assert!(ranges_per_part > 0);
-        for ranges in part.file_ranges[0].chunks(ranges_per_part) {
-            let new_part = ScanPart {
-                memtable_ranges: Vec::new(),
-                file_ranges: smallvec![ranges.to_vec()],
-                time_range: part.time_range,
+            let read_type = if compaction {
+                "compaction"
+            } else {
+                "seq_scan_files"
             };
```

Comment:
why not use `get_scanner_type` here

Reply:
It doesn't return the expected label value for this metric.